src/main/java/pro/sky/Examiner/service/JavaQuestionService.java

```diff
@@ -0,0 +1,96 @@
+package pro.sky.Examiner.service;
+
+import org.springframework.stereotype.Service;
+import pro.sky.Examiner.domain.Question;
+import pro.sky.Examiner.exception.QuestionAlreadyAddedException;
+import pro.sky.Examiner.exception.QuestionNotFoundForRemoveException;
+import pro.sky.Examiner.exception.QuestionStorageIsFullException;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Service
+public class JavaQuestionService implements QuestionService {
+    Set<Question> questions = new HashSet<>();
+    private Random random = new Random();
+
+    @Override
+    public int  getSize(){
+        return questions.size();
+    }
+
+    public JavaQuestionService() {
+        questions.add(new Question("вопрос_01", "ответ_01"));
+        questions.add(new Question("вопрос_02", "ответ_02"));
+        questions.add(new Question("вопрос_03", "ответ_03"));
+    }
+
+    private final static int MAX_SIZE = 6;//максимальное кол вопросов
+
+    private void validateSize() {
+        if (MAX_SIZE == questions.size()) {
+            throw new QuestionStorageIsFullException("Массив вопросов переполнен");
+        }
+    }
+
+    private boolean find(Question question) {
+        Set<String> getListQuestion = questions.stream()
+                .map(Question::getQuestion)
+                .collect(Collectors.toSet());
+        return getListQuestion.contains(question.getQuestion());
+    }
+
+    @Override
+    public Question add(String question, String answer) {
+        validateSize();
```

Comment:
в этом методе не должно быть никакой логики, здесь надо сделать делегацию - вызов уже написанного метода:
```suggestion
        return add(new Question(question, answer));
```

Reply:
а как проверить, а если вопрос существует? не пойму как связать тогда с  JavaController    - **как передать значение** в метод   _**private Question add(Question newQuestion)**_ ? что **надо прописать** в JavaController ?

Reply:
Не нужно ничего связывать, надо просто делать строго по заданию. В задании дано 2 метода add - который принимает объект вопроса и который принимает 2 строки - вопрос и ответ. Из контроллера вызывай любой, какой нравится. Второй использоваться будет в первом - это как раз проверка, догадается ли студент делегировать, а не дублировать код.

Reply:
проверка у тебя будет как и весь код в другом методе, в этом ты только вызываешь тот метод, а там уже всё будет